src/revoke_token_request.ts

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {StringMap} from './types';
+
+/**
+ * Represents the Token Request as JSON.
+ */
+export interface RevokeTokenRequestJson {
+  token: string;
+  token_type_hint?: string;
+}
+
+
+/**
+ * Represents a revoke token request.
+ * For more information look at:
+ * https://tools.ietf.org/html/rfc7009#section-2.1
+ */
+export class RevokeTokenRequest {
+  constructor(
+      public token: string,
+      public tokenTypeHint?: string,
```

Comment:
So `tokenTypeHint` can be one of `access_token` and `refresh_token`. So rather than leave this as a `string` define a new type. Like so:

```typescript
type TokenTypeHint = 'refresh_token' | 'access_token';
```

Then the tokenTypeHint can use this type.